src/coreclr/jit/block.cpp

```diff
@@ -853,12 +853,30 @@ Statement* BasicBlock::firstStmt() const
 //------------------------------------------------------------------------
 // hasSingleStmt: Returns true if block has a single statement
 //
+// Arguments:
+//    ignoreNops - Ignore GT_NOP statements
+//
 // Return Value:
 //    true if block has a single statement, false otherwise
 //
-bool BasicBlock::hasSingleStmt() const
+bool BasicBlock::hasSingleStmt(bool ignoreNopStatements) const
 {
-    return (firstStmt() != nullptr) && (firstStmt() == lastStmt());
+    if (firstStmt() == nullptr)
+    {
+        // No statements at all
+        return false;
+    }
+
+    const Statement* stmt = firstStmt();
+    while (ignoreNopStatements && ((stmt->GetRootNode() == nullptr) || (stmt->GetRootNode()->OperIs(GT_NOP))))
```

Comment:
Is a statement with a null node possible? Overall, do we actually find any NOPs in the return blocks?

Reply:
I wasn't sure in this, but if I remove the nullcheck it won't fail on CI so I presume it's not possible? I wish we could have a centralize "IR reference" where we could check what's legal and what's not in IR

Reply:
> Is a statement with a null node possible? Overall, do we actually find any NOPs in the return blocks?

I'll check the diffs

Reply:
It is not possible.

Reply:
Reverted these changes, there were no additional diffs from that